### PR TITLE
Add support for first line number setting

### PR DESCRIPTION
**Pull request by andrewda**
image

## Description

This adds a setting (when `Line numbers` is on) that allows the user to specify the first line number.

Closes https://github.com/carbon-app/carbon/issues/842

### CODE REVIEW

File: components/Settings.js [hunk 1]
```diff
@@ -79,6 +81,16 @@ const WindowSettings = React.memo(
           enabled={lineNumbers}
           onChange={onChange.bind(null, 'lineNumbers')}
         />
+        {lineNumbers && (
+          <Text
+            label="First line number"
+            type="int"
+            value={firstLineNumber}
+            minValue={1}
```
Comment (mfix22) [inline components/Settings.js]: Let's do `0` in case someone wants zero-based indexing 🤷‍♂

File: lib/routing.js [hunk 2]
```diff
@@ -56,6 +56,7 @@ const readMappings = [
   { field: 'pv:paddingVertical' },
   { field: 'ph:paddingHorizontal' },
   { field: 'ln:lineNumbers', type: 'bool' },
+  { field: 'fl:firstLineNumber' },
```
Comment (mfix22) [inline lib/routing.js]: Let's add a type that checks if the value is a number, or returns `undefined`

File: components/Text.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import React from 'react'
```
Comment (mfix22) [inline components/Text.js]: I'm going to touch this up a bit. Thanks for creating this though!

Comment (andrewda) [reply components/Text.js]: Looks much better now, thanks for the touch-ups!

**Comment by mfix22** [comment]
Thanks again Andrew! 🙏

**Pull request merged**